### PR TITLE
[BAD-131] - Sqoop job fails when running from Advanced mode with "argument -connect is missing" error

### PR DESCRIPTION
**Pull request by mtseu**
1. use getConnect() instead of getConnectFromAdvanced()
2. Sort out logic dependent on UI mode
3. Split configure() method to configureDatabase() and configureShim()
4. Unit test added
5. CheckStyle fixes

@mattyb149 @brosander could you please review

### CODE REVIEW

**Comment by mtseu** [comment]
@mattyb149 @brosander please ...

**Comment by mtseu** [comment]
@brosander what about this please ...

File: src/org/pentaho/di/job/entries/sqoop/AbstractSqoopJobEntry.java [hunk 1]
```diff
@@ -344,6 +355,15 @@ public void configure( HadoopShim shim, S sqoopConfig, Configuration conf ) thro
           shim.configureConnectionInformation( environmentSubstitute( nc.getHdfsHost() ), environmentSubstitute( nc.getHdfsPort() ),
               environmentSubstitute( nc.getJobTrackerHost() ), environmentSubstitute( nc.getJobTrackerPort() ), conf, messages );
         }
+/*
```
Comment (brosander) [inline src/org/pentaho/di/job/entries/sqoop/AbstractSqoopJobEntry.java]: What is this for?

File: src/org/pentaho/di/job/entries/sqoop/AbstractSqoopJobEntry.java [hunk 2]
```diff
@@ -315,24 +315,35 @@ protected void executeSqoop( HadoopShim hadoopShim, SqoopShim shim, S config, Co
...
+  }
+
+  /**
+   * Configure database connection information
+   * @param sqoopConfig - Sqoop configuration
+   */
+  public void configureDatabase( S sqoopConfig ) throws KettleException {
+    DatabaseMeta databaseMeta = getParentJob().getJobMeta().findDatabase( sqoopConfig.getDatabase() );
+
+    // if databaseMeta == null we assume "USE_ADVANCED_MODE" is selected on QUICK_SETUP
```
Comment (brosander) [inline src/org/pentaho/di/job/entries/sqoop/AbstractSqoopJobEntry.java]: So are we not setting connect info on advanced mode now?

Comment (mtseu) [inline src/org/pentaho/di/job/entries/sqoop/AbstractSqoopJobEntry.java]: @brosander it's already there - this is why we process only the case when mode is QUICK_SETUP and we were able to identify the named connection.

**Comment by brosander** [comment]
@mtseu can we remove the commented out git diff at https://github.com/pentaho/big-data-plugin/pull/347/files#diff-f8314271799bc2a6d46c44cb6b3eba89R359?

**Comment by mtseu** [comment]
@brosander Hey Bryan - the comment has been removed. do you think it could be merged now?

**Comment by mtseu** [comment]
@brosander would you mind to take a look. all your notes have been reflected

**Pull request merged**